pythia/modules/metrics.py

```diff
@@ -198,6 +199,86 @@ def calculate(self, sample_list, model_output, *args, **kwargs):
         return value
 
 
+@registry.register_metric("caption_accuracy")
+class CaptionAccuracy(BaseMetric):
```

Comment:
Is there some way it can be more generic and call handle any text strings and probabilities. I believe it requires minor changes. You will probably have to document in more detail.

Important: Please divide metrics into separate classes. This should be called `Bleu4` or `Blue4ForCaption` and you should implement other metrics separately so they are easily reusable.